crates/common/mqtt_channel/src/connection.rs

```diff
@@ -166,6 +170,22 @@ impl Connection {
                     let _ = message_sender.send(msg.into()).await;
                 }
 
+                Ok(Event::Incoming(Packet::ConnAck(ack))) => {
+                    // Workaround for  https://github.com/bytebeamio/rumqtt/issues/250
+                    // If session_name is not provided, then re-subscribe
+                    if config.session_name.is_none() {
+                        eprintln!("INFO: Connected");
+                        if let Some(err) = MqttError::maybe_connection_error(&ack) {
+                            eprintln!("ERROR: Connection Error {}", err);
+                        };
+                        let subscriptions = config.subscriptions.filters();
+                        if subscriptions.is_empty() {
+                            break;
+                        }
+                        Connection::subscribe_to_topics(&mqtt_client, subscriptions).await?;
```

Comment:
The check for an empty subscription list must be moved inside the `subscribe_to_topics` function.

```suggestion
                        Connection::subscribe_to_topics(&mqtt_client, subscriptions).await?;
```

Reply:
fixed